common/variant/src/main/java/org/apache/spark/types/variant/VariantUtil.java

```diff
@@ -377,11 +405,52 @@ public static long getLong(byte[] value, int pos) {
       case TIMESTAMP:
       case TIMESTAMP_NTZ:
         return readLong(value, pos + 1, 8);
+      case YEAR_MONTH_INTERVAL:
+        return readLong(value, pos + 2, 4);
+      case DAY_TIME_INTERVAL:
+        return readLong(value, pos + 2, 8);
       default:
         throw new IllegalStateException(exceptionMessage);
     }
   }
 
+  // Class used to pass around start and end fields of year-month and day-time interval values.
+  public static class IntervalFields {
+    public IntervalFields(byte startField, byte endField) {
+      this.startField = startField;
+      this.endField = endField;
+    }
+
+    public final byte startField;
+    public final byte endField;
+  }
+
+  // Get the start and end fields of a variant value representing a year-month interval value. The
+  // returned array contains the start field at the zeroth index and the end field at the first
+  // index.
+  public static IntervalFields getYearMonthIntervalFields(byte[] value, int pos) {
+    long fieldInfo = readLong(value, pos + 1, 1);
+    IntervalFields intervalFields = new IntervalFields((byte) (fieldInfo & 0x1),
+            (byte) ((fieldInfo >> 1) & 0x1));
+    if (intervalFields.endField < intervalFields.startField) {
```

Comment:
Do we need to check the type to be year-month interval, and the length? Some other functions call `checkIndex`.

Reply:
Yes, that would have been a good idea. `checkIndex` is performed by `readLong` anyway though. Done.